aws-android-sdk-iot/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -722,21 +727,48 @@ public void onFailure(Throwable exception) {
     }
 
     @Test
-    public void mqttWebSocket() throws Exception {
+    public void testWebSocketWithIamAuth() throws Exception {
+        websocketConnectionTest(AuthenticationMode.IAM);
+    }
+
+    @Test
+    public void testWebSocketWithCustomAuth() throws Exception {
+        websocketConnectionTest(AuthenticationMode.CUSTOM_AUTH);
+    }
 
+    private void websocketConnectionTest(AuthenticationMode authMode)
+            throws InterruptedException, JSONException {
         final ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus> statuses = new ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus>();
         final ArrayList<String> messages = new ArrayList<String>();
 
-        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-w-ws", Region.getRegion(Regions.US_EAST_1), endpointPrefix);
+        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-w-ws",
+                Region.getRegion(Regions.US_EAST_1), endpointPrefix);
 
         mqttManager.setAutoReconnect(false);
-        // connect using WebSockets and IAM credentials
-        mqttManager.connect(credentialsProvider, new AWSIotMqttClientStatusCallback() {
-            @Override
-            public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
-                statuses.add(status);
-            }
-        });
+
+        if (AuthenticationMode.CUSTOM_AUTH.equals(authMode)) {
```

Comment:
```
switch (authMode) {
    case CUSTOM_AUTH:
}
```

In the `if`, you use just `AuthenticationMode`, but in the `else if`, you use `AWSIotMqttManager.AuthenticationMode`. I'd pick one and stick with it. So long as you can use the second one without getting any very long lines, I'd probably use it.

Reply:
Updated with `AWSIotMqttManager.AuthenticationMode`. I agree that this is more descriptive and preferable.